### DISCUSSION TITLE
RangeError: Invalid time value

### DISCUSSION THREAD

**Discussion by MickaelMenet** [js-fns/date-fns#2928] [Q&A] [answered]
Hello,

I'm trying to convert the following code from moment to date-fns and I get the following message: "RangeError: Invalid time value". Can you help me ?

import moment from 'moment';
<Typography>{`${bla} ${moment(myDate).format('DD/MM/YYYY')}`}</Typography>

{console.log(myDate)}
Result =>
2022-01-14T07:57:57.288+0000

**Conversion test to date-fns:**

import { format } from 'date-fns';

<Typography>{`${bla} ${format(new Date(myDate), 'dd/MM/yyyy')}`}</Typography>

result =>

 370 | var originalDate = toDate(dirtyDate);
  371 |
  372 | if (!isValid(originalDate)) {
> 373 | throw new RangeError('Invalid time value');
      | ^ 374 | } // Convert the date in system timezone to the same date in UTC+00:00 timezone.
  375 | // This ensures that when UTC functions will be implemented, locales will be compatible with them.
  376 | // See an issue about UTC functions: https://github.com/date-fns/date-fns/issues/376

**Comment by fturmel** [comment]
Your conversion seems OK to me, I can't reproduce the problem:

```js
import moment from "moment";
import { format } from "date-fns";

const myDate = "2022-01-14T07:57:57.288+0000";

const r1 = moment(myDate).format("DD/MM/YYYY");
console.log(r1); // 14/01/2022

const r2 = format(new Date(myDate), "dd/MM/yyyy");
console.log(r2); // 14/01/2022
```

The error message you are getting implies that you are not passing a valid input in the Date constructor, so maybe double check that `myDate` really is a string, and in a compatible format. Isolate `new Date(myDate)` to troubleshoot.

<img width="424" alt="Screen Shot 2022-01-25 at 8 57 46 AM" src="https://user-images.githubusercontent.com/4009209/150990331-fd99ab3b-a051-4e06-8006-3f64271c3b67.png">

**Reply by MickaelMenet** [reply]
It is indeed a character string, everything seems valid. I tested in an isolated file it works ... So I do not understand where my problem comes from. Thanks for your help ;)

**Reply by fturmel** [reply]
No problem. Please update the community on your issue, it might help someone down the road.

A few more ideas for you:
- Type coercion could cause something that is not a string to output as a string. Try `typeof myDate` to make sure.
- Consider using a debugger if you can't get to the bottom of it, it'll be much easier to inspect your application state that way.
- Again, isolate `new Date(myDate)` and find that bad input that creates an invalid date

Good luck!

**Reply by MickaelMenet** [reply]
I may have a clue but it seems strange to me and I don't really see how to solve it. When I console.log the result of my date I have undefined several times before it is logged. 
Moment finally manages to manage this on its own but not date-fns because it must consider that the undefined is false .

![image](https://user-images.githubusercontent.com/71713999/151197554-38b0f01c-f647-4ce5-9f84-68136448e2e1.png)

**Reply by fturmel** [reply]
Yes, `new Date(undefined)` is invalid time and the `format` method throws an Error which interrupts the execution. `moment(undefined)` seems to be interpreted as current time and doesn't fail.

Just wait until `myDate` is not undefined to create the Date instance and call `format`.

**Comment by MickaelMenet** [comment, marked as answer]
I just managed this with a ternary for now while waiting for the date of the date to load nicely. I'll come back to that later. What emerges from my problem date fns does more control than moment especially on the structure of the date received!
